vim-patch:7.4.1546, 7.4.1548

prollings opened the pull request:
vim-patch:7.4.1546

Problem:    Sticky type checking is more annoying than useful.
Solution:   Remove the error for changing a variable type.

https://github.com/vim/vim/commit/f6f32c38bf3319144a84a01a154c8c91939e7acf

Note: There are a bunch of other changes to eval.txt that I believe are
N/A and not related to this patch.

vim-patch:7.4.1548

Problem:    Two tests fail.
Solution:   Adjust the expected error number. Remove check for type.

https://github.com/vim/vim/commit/5a2800fd141a8fc0c80cdf421dcb76001a22327f

Commit:
vim-patch:7.4.1546

Problem:    Sticky type checking is more annoying than useful.
Solution:   Remove the error for changing a variable type.

https://github.com/vim/vim/commit/f6f32c38bf3319144a84a01a154c8c91939e7acf

Note: There are a bunch of other changes to eval.txt that I believe are
N/A and not related to this patch.

KillTheMule commented on src/nvim/testdir/Makefile:
```diff
@@ -33,9 +33,11 @@ SCRIPTS := \
 	    test_cursor_func.res \
 	    test_hardcopy.res \
 	    test_help_tagjump.res \
+	    test_history.res \
```
This is in your other PR :)

prollings replied:
Haha, interesting... :P

KillTheMule commented:
Test integration looks fine to me

KillTheMule commented:
The tests need to be adjusted, see https://github.com/vim/vim/commit/5a2800fd141a8fc0c80cdf421dcb76001a22327f. Maybe also https://github.com/vim/vim/commit/3905e291fe4375ca5c59efa9ffcb01a39c7be3a9.

prollings commented:
@KillTheMule That makes sense. I'll package in those patches to this PR.

Commit:
vim-patch:7.4.1548

Problem:    Two tests fail.
Solution:   Adjust the expected error number. Remove check for type.

https://github.com/vim/vim/commit/5a2800fd141a8fc0c80cdf421dcb76001a22327f

KillTheMule commented:
In that case, you should relabel this PR as [WIP].

justinmk commented on src/nvim/testdir/Makefile:
```diff
@@ -33,6 +33,7 @@ SCRIPTS := \
 # Tests using runtest.vim.vim.
 # Keep test_alot*.res as the last one, sort the others.
 NEW_TESTS = \
+	    test_assign.res \
```
This should be in `test_alot.vim`. I fixed it and others that were here.
